torch_geometric/explain/algorithm/base.py

```diff
@@ -37,6 +38,11 @@ def forward(
             target (torch.Tensor): the target of the model.
             target_index (int or torch.Tensor, optional): The target indices to
                 explain. (default: :obj:`None`)
+            index (Union[int, Tensor], optional): the node/edge index
```

Comment:
In my mind it would be used like this: if you have a model that outputs a $y \in \mathbb{R}^K$ with $K>1$ for all your nodes, and you are interested in what influences $y[k]$ for a specific node $i$, you would set `target_index=k` and `index=i`. The target index influences the index of the output (regardless of the task level of the GNN), while the `index` helps selecting the specific node (and edge in the future).

Reply:
Oh, I see. `target_index` references a second dimension of potentially multiple targets. I thought it would index in the first dimension (and therefore didn't see the difference to `index`). Wonder if it might be better to allow tuple of indices then?

Reply:
Yes, exactly (I might still this comment to put it in the docs) ! 
Perhaps it would be confusing ? I find it a bit clearer if they are separated (maybe renaming  `index` to `node_index` might help.

Reply:
Sounds good.